### ISSUE TITLE
Function/method calls after is_callable check not properly evaluated

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#97]
Right now Psalm gives up on checking function parameters match the given function after an `is_callable` check, so this issue isn't currently caught (false negative):

```php
/**
 * @param callable $callback
 * @return void
 */
function route($callback) {
  if (!is_callable($callback)) {  }
  takes_int("string");
}

function takes_int(int $i) {}
```

Instead of turning off all function parameter checks, Psalm should just stop checking if a function exists (but still proceed normally for functions that do exist).
